plugins/processors/gpuattributes/processor_test.go

```diff
@@ -19,93 +19,170 @@ func TestProcessMetrics(t *testing.T) {
+				},
 			}),
 			wantCnt: 1,
-			want: map[string]string{
-				"ClusterName": "cluster",
+			want: []map[string]string{
+				{
+					"ClusterName": "cluster",
+				},
 			},
 		},
 		"nodeDropJson": {
-			metrics: generateMetrics("node", map[string]string{
-				"ClusterName": "cluster",
-				"kubernetes":  "{\"host\":\"test\"}",
+			metrics: generateMetrics("node", []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"kubernetes":  "{\"host\":\"test\"}",
+				},
 			}),
 			wantCnt: 1,
-			want: map[string]string{
-				"ClusterName": "cluster",
-				"kubernetes":  "{\"host\":\"test\"}",
+			want: []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"kubernetes":  "{\"host\":\"test\"}",
+				},
 			},
 		},
 		"nodeDropMixed": {
-			metrics: generateMetrics("node", map[string]string{
-				"ClusterName": "cluster",
-				"Drop":        "val",
-				"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+			metrics: generateMetrics("node", []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"Drop":        "val",
+					"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+				},
 			}),
 			wantCnt: 1,
-			want: map[string]string{
-				"ClusterName": "cluster",
-				"kubernetes":  "{\"host\":\"test\"}",
+			want: []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"kubernetes":  "{\"host\":\"test\"}",
+				},
 			},
 		},
 		"dropPodWithoutPodName": {
-			metrics: generateMetrics("pod", map[string]string{
-				"ClusterName": "cluster",
-				"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+			metrics: generateMetrics("pod", []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+				},
 			}),
 			wantCnt: 0,
-			want:    map[string]string{},
+			want:    []map[string]string{},
 		},
-		"keepPodWithoutPodName": {
-			metrics: generateMetrics("pod", map[string]string{
-				"ClusterName": "cluster",
-				"PodName":     "pod",
-				"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+		"keepPodWithPodName": {
+			metrics: generateMetrics("pod", []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"PodName":     "pod",
+					"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+				},
 			}),
 			wantCnt: 1,
-			want: map[string]string{
-				"ClusterName": "cluster",
-				"PodName":     "pod",
-				"kubernetes":  "{\"host\":\"test\"}",
+			want: []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"PodName":     "pod",
+					"kubernetes":  "{\"host\":\"test\"}",
+				},
 			},
 		},
 		"dropContainerWithoutPodName": {
-			metrics: generateMetrics("container", map[string]string{
-				"ClusterName": "cluster",
-				"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+			metrics: generateMetrics("container", []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+				},
 			}),
 			wantCnt: 0,
-			want:    map[string]string{},
+			want:    []map[string]string{},
```

Comment:
why are we changing this from a map to a list of maps?

Reply:
It turns out that each data point can have their own attribute sets. `want` now represents attribute sets of each data points in a metric object.

Reply:
ahhh got it!